test/odbc/main.cpp

```diff
@@ -1,8 +1,26 @@
 #include <gtest/gtest.h>
+#include <map>
+#include <string>
 #include "odbc_handler.h"
 
+using std::map;
+using std::string;
+
 int main(int argc, char **argv) {
+  OdbcHandler odbcHandler;
+  map<ServerType, ConnectionStringObject> drivers = odbcHandler.getOdbcDrivers();
+  string filter_string = "";
 
+  map<ServerType, ConnectionStringObject>::iterator it;
+  for (it = drivers.begin(); it != drivers.end(); it++)
+  {
+    std::cout<<it->second.GetConnectionString()<<"\n";
+    filter_string.append(it->second.GetTestToRun());
+    filter_string.append("*:");
```

Comment:
Not sure if we've explored this part yet, but for the tests that will be switching between both drivers, would we use a different prefix for the TEST_TO_RUN field
and require that both types of drivers are used with the prefix name. 

But one thing to think about is if we do decide to have a test that uses both drivers, we can't call them `PSQL_MSSQL_testname` or `MSSQL_PSQL_testname` since these tests would run if any of the drivers are initialized.

Reply:
I'm still thinking about how to run tests that use multiple drivers. For 2 drivers it's easy, just check if both drivers exist & if they do then run the tests that switch between the 2 drivers.

But for this test suite to be fully extensible in cases where there are more than 2 drivers, it becomes trickier. If we had drivers `A` `B` & `C` we would need to be able to run tests that test both drivers `A` & `B`, `B` & `C`, `A` & `C`, etc. Becomes a classic combinations/permutations problem. 
  
Good point about the test fixture name if we have tests using both drivers. We will have to be careful about naming the test fixture, especially when we have more than 2 drivers.

Reply:
Actually, one idea is instead of main.cpp determining which tests can be run depending on the driver(s) provided, we can move this logic to the test fixtures themselves. For example:

```
class PSQL_Connection : public testing::Test {
  void SetUp() override {
   OdbcHandler odbcHandler;
    if (PSQLODBC not in odbcHandler.getOdbcDrivers())
      GTEST_SKIP() << "Skipping all tests for this fixture";
  }
  ...
};
```
While this provides greater flexibility in determining if the test fixture can be run or not (all it takes is an `if` statement) there a couple cons:

1) This logic will need to be in every test fixture. I don't think there's a way to batch-apply similar logic to all test fixtures within a certain directory. 

2) Have to create an `OdbcHandler` object for every test fixture SetUp() in order to get the list of available drivers. 

I'll have to test this out, but this is an idea.

Reply:
I like this idea because then the filters wouldn't be name-dependent be name-dependent. 

Con 1: Yes, this would be unfortunate but I think the benefit of controlling which tests are ran based on driver availability is better. Also, I think this might be better since if (for some reason) we wanted to add GTEST flags through the terminal command line (I assume the current method if appending "PSQL*:" conflicts with filtering tests through the command lines which I believe GTEST supports if im not wrong?). O

Con 2: This can be handled with that new class that handles the config stuff. Alternatively, main.cpp could also set some global boolean variables or other config  variables  so we can simply check if a driver exists.

Reply:
Thanks, if there are no objections moving the logic to the test fixtures is the way I'll go. 

Could you clarify what you meant by a couple things in Con2:

> This can be handled with that new class that handles the config stuff.

Yes, but wouldn't I still need to create this NewClass object in every test fixture SetUp() still to get the list of available ODBC drivers? Unless you're suggesting some global NewClass object all test fixtures can access. I'm not sure what the proper code etiquette would be here in this case, whether to use global or not.

> Alternatively, main.cpp could also set some global boolean variables or other config variables so we can simply check if a driver exists.

So if we have `n` drivers we have `n` global bool variables? Not sure how I feel about that. I may be misinterpreting what you mean here.

Reply:
> Yes, but wouldn't I still need to create this NewClass object in every test fixture SetUp() still to get the list of available ODBC drivers? Unless you're suggesting some global NewClass object all test fixtures can access. I'm not sure what the proper code etiquette would be here in this case, whether to use global or not.

My bad, I was thinking more of a global class that handles configuration or a static class (but static classes don't exist in C++).  I think global might be fitting because information about the config file and environment variable should persist through the different tests and should not change. So having a global state would be preferred than having to re-initialize the same object to produce the same information. 

> So if we have `n` drivers we have `n` global bool variables? Not sure how I feel about that. I may be misinterpreting what you mean here

The global boolean variable is just an alternative to having a global class. I don't think scaling would be an issue since the current design only handles 2 drivers at most (1 pssql and 1 mssql and even if we add a 3rd one, we might need to change code in other places as well).

Reply:
```
class PSQL_Connection : public testing::Test {
  void SetUp() override {
   OdbcHandler odbcHandler;
    if (PSQLODBC not in odbcHandler.getOdbcDrivers())
      GTEST_SKIP() << "Skipping all tests for this fixture";
  }
  ...
};
```

If we were to go this route, why not instead of hardcoding what to skip, 
iterate through the odbcHandler.getOdbcDrivers() and execute the tests for all drivers that are present?

Reply:
Currently we're planning to execute all the related tests. As a developer, I would like to be able to specify just one text fixture or some. Even if we don't add this functionality in the first iteration, we should think how hard would it be given our current approach.

Reply:
>As a developer, I would like to be able to specify just one text fixture or some

@leszekamz From my understanding, this is built into google test. It's possible to execute just one test fixture by using the `--gtest_filter=TEST_NAME_TO_RUN`  flag when running the executable. 

However, the state of the current PR which calls `GTEST_FLAG(filter)` may conflict with this flag which is why the proposed route of handling the logic within the setup test fixture setup may be a better option

Reply:
> As a developer, I would like to be able to specify just one text fixture or some.

I think I have an idea of how to do this. Much like `jdbc_schedule` I will have an `odbc_schedule` file. Developer can specify test fixture(s) in that file to specifically run. I will parse through this file & using the googletest framework I will be able to run only those fixtures specified.

Question now becomes what if developer has specified a fixture to run (like PSQL_Connection) but the ODBC driver required by that fixture (like psqlODBC) is not provided? Should we let the tests under that fixture fail? Should we not run the tests at all & just print out a message saying the tests were not run due to the specific ODBC driver being missing? This all leads back to the question of how do we determine the tests to run depending on which ODBC drivers are provided.

Reply:
Added `odbc_schedule` file and went with adding the test skipping logic to the test fixtures.